cpp/src/data_type.cc

```diff
@@ -42,7 +42,8 @@ std::shared_ptr<arrow::DataType> DataType::DataTypeToArrowDataType(
   case Type::LIST:
     return arrow::list(DataTypeToArrowDataType(type->child_));
   default:
-    throw std::runtime_error("Unsupported data type");
+    std::string msg = "The data type " + type->ToTypeName() +
```

Comment:
I wonder how the default case return or throw, because it doesn't `return` or `throw` directly.

Reply:
throw std::runtime_error(msg);

Reply:
Thanks for pointing out, fixed.